### ISSUE TITLE
GSDX11 Hardware bug: Xenosaga Episode III shadow bug

### ISSUE THREAD

**Issue by BlackTelomeres** [PCSX2/pcsx2#1730]
pcsx2-v1.5.0-dev-1749 (12-31-16)
Plugins: GSDX AVX2 and the defaults
DX11 renderer - native res and up (Hardware mode only)
CRC Hack Level: Full (safest)
Not really present in Software mode or DX9 hardware mode

To Replicate: Start a new game, Skip the cutscenes and begin walking forward once you gain control of the player character.

https://s29.postimg.org/g03yr3b47/gsdx9_vs_gsdx11_xenosaga_3.jpg

A large black box appears on the ground and moves/changes in relation to the player character (Shion) in the first room in the game that you gain control of a character in. Seems to have something to do with the shadow system of the game is my best guess. At some points, depending on where Shion is in the room, it will disappear but mainly it will appear where the room turns from a larger area to a smaller corridor.

The "alpha" and "alpha stencil" hacks do not fix this bug. Brief experimentation with the "TC Offset" hacks did not seem to fix it. A skipdraw of 99 did not remove the box either although I would've thought it would have.

Strangely enough, the "half pixel offset" hack--which is only available when playing above native res--alters the look of, although does not fix, the bug. It also slightly changes when it appears. It then appears behind Shion on the right rather than in front of her on the left. Here is the bug with the half pixel offset on:

https://s23.postimg.org/y125e1rmj/xenosaga_3_pcsx2_dx11_with_half_pixel_offset.jpg

What is interesting is that the shadows display slightly differently in the DX9 renderer and the bug is not really present in that renderer with no other settings changed besides the renderer itself. The shadow of Shion looks slightly different in DX9 (has sort of a dark outline) which may have something to do with why the bug is not appearing in that renderer, The DX9 renderer not having the issue may hint at what is causing it or how to fix it in DX11.

If one turns off "alpha correction" in DX9 (it is on by default), something similar but not exactly the same will happen with a giant grey box (a transparent shadow effect) appearing under Shion in place of her shadow.

I have not played further to see how many other areas the bug is present in.

**Comment by refractionpcsx2** [comment]
Having crc hack set to full may not be helping you there

**Comment by BlackTelomeres** [comment]
"Partial CRC hacks" or lower fixes it for that room but then it reappears in the next room anyway. Also adds some ugly bloom blurring to the edges of the screen if above native res, but that's a separate issue I suppose.

https://s24.postimg.org/3optsmcn9/xenosaga_3_room_2_partial.jpg

**Comment by gregory38** [comment]
@BlackTelomeres Did you the test the OpenGL renderer ? Various features aren't supported on Dx.

**Comment by angreis** [comment]
hello everyone i solved this problem of black box in the ground,
i use the internal resolution custom and put 1680x1050, you can put 3360x2100 it works

**Comment by MrCK1** [comment]
@BlackTelomeres Any status change on the issue if you use the latest dev build?

**Comment by lightningterror** [comment]
Recent fbmask port should render the effect properly if blending on d3d is set to high but it will be super slow, I'm afraid that's the best we can do atm.

See https://github.com/PCSX2/pcsx2/pull/2937

**Comment by BlackTelomeres** [comment]
Thanks for reopening this. To respond to some of the previous questions: if no hacks are applied, then yes XS3 shadows are also a current issue in OGL HW as of 1.5.0-dev-3308 (though at the time this issue was started - 2017 - I wasn't able to use OGL HW well so I had not tested OGL HW until 2019). 

The "Merge Sprite" hack allows character shadows to appear in OGL HW (works/needed even at native res). The "disable safe features" hack is also needed to get rid of the black "shadow box" on the ground that also occurs beside the character if upscaling (see the image in the GSdump zip later in this post for what it tends to look like). Sorry I didn't check back sooner to see the questions.

I tested angreis's suggestion of a specific custom resolution 1680x1050, and it causes the black box error to not appear in OGL so it can substitute for needing the hack "disable safe features" - but also doesn't show any shadows by my testing without also enabling the "merge sprite" hack. 

So basically you need either "Merge sprite" and 1680x1050 custom res or "Merge sprite" and "disable safe features" for town/dungeon shadows to look right when upscaling.

Also perhaps worth noting that "preload frame data" and "Half Pixel Offset-Special (texture)" - to mitigate a bloom alignment issue when upscaling - were two other hacks I found basically needed for XS3 to allow most of the game to look correct on OGL HW. There are a few other hiccups specific to certain scenes.

I attached a GSdump relevant to shadows in case it is needed

[xenosaga3 shadow error without mergesprite.zip](https://github.com/PCSX2/pcsx2/files/3902638/xenosaga3.shadow.error.without.mergesprite.zip)

**Comment by Softernet** [comment]
Note: OpenGL renders do not have this problem on v1.5.0-dev-2658 and later, but in v1.5.0-dev-2858 and later.

**Comment by gugylucky** [comment]
just want to report that with the new vulkan renderer in v1.7.2208 the black box error is back even with the "disable safe features" hack.
[xs3 with mergesprite and disable safe features.zip](https://github.com/PCSX2/pcsx2/files/7833056/xs3.with.mergesprite.and.disable.safe.features.zip)

**Comment by refractionpcsx2** [comment]
so it's fine in 2207?

**Comment by gugylucky** [comment]
ah sorry, was meaning to say it's broken on Vulkan in general, but fine on OpenGL HW

**Comment by refractionpcsx2** [comment]
okay just for clarification, is it broken in DX (as related to the original bug which you said was back), or is it only broken in vulkan?

**Comment by gugylucky** [comment]
yeah that was my bad. it's only broken in vulkan

**Comment by refractionpcsx2** [comment]
D3D 11 seems to be broken with your dump, also

**Comment by AsmodeanBreaggAI** [comment]
After some experimentation on Direct 3D11, myself and a friend managed to find fixes for some of the game's issues at least. I'm using PCSX2 Nightly v.1.7.2746.
In Graphics Settings > Hacks > Skipdraw range, change both boxes to 3. This removes the strange square shadow issue following Shion around. I've seen some builds recommend CRC Hack Level to aggressive for this game, but make sure it is set to Automatic (Default) instead in Graphics Settings > Renderer, or the shadow issue will remain.
In Graphics Settings > Hacks > Half-Pixel Offset, change it to Normal (Vertex). This will remove the odd distortions around characters in cutscenes.
In Graphics Settings > Renderer > Blending Accuracy, change it to High. This may reduce performance if your CPU is on the lower end, but it resolves the issue of the screen randomly becoming darker when moving through certain areas or text-based scenes.
